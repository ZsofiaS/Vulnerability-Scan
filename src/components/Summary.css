.Summary {
  margin: 2rem 8rem;
  padding: 1rem 2rem;
  text-align: center;
  width: 100%;
}
.summaryContainer {
  display: flex;
  justify-content: space-around;
  width: 100%;
  text-align: left;
}
.status, .assets, .severities {
  border: 1px solid #F2F3F4;
  padding: 1rem;
  background-color: #F4F6F6;
  flex-basis: 100%;
  margin: 1rem;
}
.assets {
}
.assets ul, .severities ul {
  padding: 0;
}
.asset {
  list-style-type: none;
}
.severities {
}
