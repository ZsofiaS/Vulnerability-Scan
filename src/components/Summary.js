import React from 'react';
import Moment from 'react-moment';
import SeverityItem from './SeverityItem';
import '../styles/App.css';
import data from '../data/scan';

const Summary = () => {

  const { assets, severityCounts, name, status, dateStarted } = data.scan;

  const showAssets = () => {
    return (
      assets.map((asset, i) => {
        return (
          <li
            className="asset"
            key={i}>
            {asset.description} ({asset.name})
          </li>
        );
      })
    );
  };

  const showSeverities = () => {
    return (
      Object.keys(severityCounts).map((key, i) => {
        return (
          <SeverityItem
            key={i}
            severity={key}
            count={severityCounts[key]}
          />
        );
      })
    );
  };

  const convertDate = (date) => {
    return (
      <Moment format='Dd MMM HH:mm'>{date}</Moment>
    );
  };

  return (
    <div className="summary">
      <h1>{name}</h1>
      <h2>Summary</h2>
      <div className='summaryContainer'>
        <div className='status'>
          <p>Status: {status}</p>
          <p>Scan started: {convertDate(dateStarted)}</p>
        </div>
        <div className='assets'>
          <p>Scanned: </p>
          <ul>
            { showAssets() }
          </ul>
        </div>
        <div className='severities'>
          <p>Severities found:  </p>
          <ul>
            { showSeverities ()}
          </ul>
        </div>
      </div>
    </div>
  );
};

export default Summary;
