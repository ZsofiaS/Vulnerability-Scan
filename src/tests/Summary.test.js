import React from 'react';
import { shallow } from 'enzyme';
import Summary from '../components/Summary';
import toJson from 'enzyme-to-json';

describe('renders Summary component', () => {
  it('renders without crashing', () => {
    shallow(<Summary />);
  });
  it('renders header', () => {
    const wrapper = shallow(<Summary />);
    const header = (<h2>Summary</h2>);
    expect(wrapper.contains(header)).toEqual(true);
  });
});

describe('snapshot', () => {
  it('Summary snapshot', () => {
    const tree = shallow(<Summary />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
