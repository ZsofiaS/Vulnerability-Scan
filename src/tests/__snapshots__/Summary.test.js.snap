// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot Summary snapshot 1`] = `
<div
  className="summary"
>
  <h1>
    Monthly Vulnerability Scan
  </h1>
  <h2>
    Summary
  </h2>
  <div
    className="summaryContainer"
  >
    <div
      className="status"
    >
      <p>
        Status: 
        completed
      </p>
      <p>
        Scan started: 
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element={null}
          filter={[Function]}
          format="Dd MMM HH:mm"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          2020-01-01T00:00:00.000Z
        </t>
      </p>
    </div>
    <div
      className="assets"
    >
      <p>
        Scanned: 
      </p>
      <ul>
        <li
          className="asset"
          key="0"
        >
          Email server
           (
          T09277
          )
        </li>
        <li
          className="asset"
          key="1"
        >
          Main Database
           (
          T09320
          )
        </li>
        <li
          className="asset"
          key="2"
        >
          Firewall
           (
          T09746
          )
        </li>
      </ul>
    </div>
    <div
      className="severities"
    >
      <p>
        Severities found:  
      </p>
      <ul>
        <SeverityItem
          count={0}
          key="0"
          severity="critical"
        />
        <SeverityItem
          count={2}
          key="1"
          severity="high"
        />
        <SeverityItem
          count={2}
          key="2"
          severity="medium"
        />
        <SeverityItem
          count={1}
          key="3"
          severity="low"
        />
        <SeverityItem
          count={2}
          key="4"
          severity="information"
        />
      </ul>
    </div>
  </div>
</div>
`;
